src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -548,7 +553,7 @@ public void removeElementFilesAndLangKeys(ModElement element) {
 				if (conditionRaw != null || GeneratorTokens.containsVariableTokens(rawname)) {
 					if (generatableElement == null) {
 						generatableElement = element.getGeneratableElement();
-						if (generatableElement == null) {
+						if (generatableElement == null && performFSTasks) {

```

Comment:
You should add new flag logWarnings here, as this is not at all related to FS tasks